src/lsdb/core/crossmatch/abstract_crossmatch_algorithm.py

```diff
@@ -51,12 +54,22 @@ def __init__(
         self.right_pixel = right_pixel
         self.left_metadata = left_metadata
         self.right_metadata = right_metadata
+        self.right_margin_hc_structure = right_margin_hc_structure
         self.suffixes = suffixes
 
     @abstractmethod
     def crossmatch(self) -> pd.DataFrame:
         """Perform a crossmatch"""
 
+    @abstractmethod
+    def validate(self):
```

Comment:
Should they both be `(self, *args, **kwargs)` or something like that?

Reply:
Added kwargs in a few places, until pylint/mypy stopped complaining.